src/commonMain/kotlin/io.github.nomisrev/JsonPath.kt

```diff
@@ -65,24 +65,78 @@ public inline val Optional<JsonElement, JsonElement>.`null`: Optional<JsonElemen
 public inline val Optional<JsonElement, JsonElement>.every: Every<JsonElement, JsonElement>
   inline get() = this compose Every.jsonElement()
 
-/** Select property with [name] */
+/**
+ * Select value at [selector]. The following syntax is supported for [selector]:
+ * - without square brackets: select the property with that name,
+ * - `['field']`: select the property with that name,
+ * - `[i]`, where `i` is a number: select the index in an array.
+ */
 public fun Optional<JsonElement, JsonElement>.select(
-  name: String
-): Optional<JsonElement, JsonElement> =
-  `object` compose Index.map<String, JsonElement>().index(name)
+  selector: String
+): Optional<JsonElement, JsonElement> {
+  val inBrackets = matchNameInBrackets(selector)
+  val ix = matchIndexInBrackets(selector)
+  return when {
+    inBrackets != null -> `object` compose Index.map<String, JsonElement>().index(inBrackets)
+    ix != null -> get(ix)
+    else -> `object` compose Index.map<String, JsonElement>().index(selector)
+  }
+}
 
 /**
- * Select _path_ with _dot (.) notation_
+ * Select values at [selector]. The following syntax is supported for [selector]:
+ * - without square brackets: select the property with that name,
+ * - `['field']`: select the property with that name,
+ * - `*`: select all the fields or indices,
+ * - `[i]`, where `i` is a number: select the index in an array,
+ * - `[i,j,...]` where `i,j,...` are numbers: select the indices in an array,
+ * - `[start:end]`: select the indices from `start` to (but not including) `end`,
+ * - `[start:]`: select the indices from `start` to the end of the array.
+ */
+public fun Optional<JsonElement, JsonElement>.selectMultiple(
```

Comment:
Nit: wondering if we should use `Every` in the name which is already used in the API. This seems to be more in line with the already existing syntax, and the returned type `Every`.

`JsonPath.selectEvery("streets.*.name")` vs `JsonPath.select("streets").every.select("name")`

Reply:
Done!